### DISCUSSION TITLE
WalletConnect V3 Redirect issue

### DISCUSSION THREAD

**Discussion by VasilVR** [WalletConnect/walletconnect-monorepo#3929] [Developer Support]
Hello WalletConnect team!

We integrated WalletConnect V3 in our staking dApp. The problem is some of the users start to face issues using it. For example, if they want to use SafePal and choose it from the WalletConnect modal they get this error:

![photo_2023-11-15_11-05-22](https://github.com/WalletConnect/walletconnect-monorepo/assets/141453270/61981579-0cbd-4965-adbf-1ec2e5c0d481)

Besides that, WalletConnect returns this error for the BSC network (the network we are using in our staking dApp):

`Uncaught (in promise) ChainDoesNotSupportContract: Chain "BNB Smart Chain" does not support contract "ensUniversalResolver"`

Screenshot:

![Screenshot_1](https://github.com/WalletConnect/walletconnect-monorepo/assets/141453270/00460666-1b47-4f29-846b-9dcbbb599907)

I think V3 has problems with some authentications that use the `isAuth()` method for verification of the blockchain. 

Can you please take a look and let me know what steps we could perform to fix the error, so we can keep using WalletConnect V3 within our dApp.

Thank you in advance!

**Comment by glitch-txs** [comment]
Hi, are you able to reproduce this [here](https://lab.web3modal.com/)?

Also are you calling any ENS function?

**Comment by safepal-support-16313090221** [comment]
Same issues I filled in with ..I called this 1631 309 0221 number. They charged me 20 USD and they solved my issues. I guess the of the guy was Alby . Trusted people .
